### PR TITLE
User Profile Page + Edit Profile Page

### PR DESCRIPTION
**Pull request by jinfante708**
apologies, my prettier extension reformatted a lot of the code, i can turn it off if it ends up being a nuisance

data/users.js
-implemented `getUserStatistics()` to grab 6 pieces of data for a particular user (This is assuming that the `tasks` and `tasklists` collections have `userId` as a parameter in their schema now)

routes/user.js
-implemented `GET /profile` which passes in `request.session.user` which are user details (@AnkitKachhadiya let me know if I should change my try/catch block here, I did my best to follow your coding style)
-implemented `GET /update-profile` which simply brings up the form to update profile

/views/users
-created `profile.handlebars` for the profile page, its just two bootstrap card elements, one card has the user's name, email, and date of birth, along with a button to bring them to the form to update their profile, another card shows their statistics
- created 'update-profile.handlebars` for the form to update profile, users can update their `first name`, `last name`, and `date of birth`, used an AJAX request here

/public/js/editProfile.js
-did error checking on client-side for the three parameters mentioned in previous bullet
-sent AJAX request to `PUT /users/profile` to successfully update profile

i wasn't able to thoroughly test this since its dependent on the data functions for `tasks` and `tasklists` to be updated to include userId in its parameters, but this should work out of the box once it is updated, let me know if there's any changes i should make

### CODE REVIEW

File: data/users.js [hunk 1]
```diff
@@ -7,470 +7,519 @@ const bcryptjs = require("bcryptjs");
...
+      numTasks: numTasks,
+      completedOnTime: completedOnTime,
+      notCompletedOnTime: notCompletedOnTime,
+      completed: completed,
+      notCompleted: notCompleted,
+    };
+
+    return userStatistics;
+  }
+  try {
```
Comment (AnkitKachhadiya) [inline data/users.js]: `try{}` block is empty. Please put entire code inside `try{}`

File: routes/users.js [hunk 2]
```diff
@@ -8,385 +8,415 @@ const usersData = data.users;
...
-    request.app.locals.passwordUpdatedFlashMessage = undefined;
+  let user = request.session.user;
+  let userStatistics;
 
-    response.render("users/change-password", {
-        pageTitle: "Change Password",
-        passwordUpdatedFlashMessage: passwordUpdatedFlashMessage,
+  try {
+    userStatistics = await usersData.getUserStatistics(user._id);
+
```
Comment (AnkitKachhadiya) [inline routes/users.js:161]: Please check `userStatistics` is not empty

File: public/js/editProfile.js [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+    if (!hasErrors) {
+      var requestConfig = {
+        method: "PUT",
+        url: `/users/profile`,
+        contentType: "application/json",
+        data: JSON.stringify(userInfo),
+      };
+
+      $.ajax(requestConfig).then(function (responseMessage) {
+        console.log(responseMessage);
```
Comment (AnkitKachhadiya) [inline public/js/editProfile.js:98]: What will happen when `ajax` call is completed

**Review by AnkitKachhadiya — COMMENTED** [review]
I have made some comments please have a look
Overall it looks good

File: routes/users.js (hunk 2 above)
Comment (jinfante708) [reply routes/users.js:161]: Added an empty object check

File: public/js/editProfile.js (hunk 3 above)
Comment (jinfante708) [reply public/js/editProfile.js:98]: redirected user back to profile page to see their newly updated info, thanks for catching that

File: data/users.js (hunk 1 above)
Comment (jinfante708) [reply data/users.js]: whoops, code moved around when i pulled main branch into dev branch, thanks for catching that

**Pull request merged**